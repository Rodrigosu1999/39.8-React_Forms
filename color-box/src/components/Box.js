import React from "react";
import "../styles/Box.css"

const Box = ({color, height, width, id, deleteBox}) => {
    const cardStyle = {
        backgroundColor: color,
        height: height,
        width: width
    }

    return (
        <div className="Box" style={cardStyle} id={id}>
            <button onClick={deleteBox}>X</button>
        </div>
    );
}

export default Box;