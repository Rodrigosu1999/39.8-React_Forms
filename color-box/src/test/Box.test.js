import React from 'react';
import { getByText, render, screen } from '@testing-library/react';
import Box from '../components/Box';

it('Renders App', () => {
  render(<Box />);
});

it('Passes Snapshot', () => {
  const {asFragment} = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders correct text', () => {
  const {getByText} = render(<Box />);
  console.log(getByText('X'))
});