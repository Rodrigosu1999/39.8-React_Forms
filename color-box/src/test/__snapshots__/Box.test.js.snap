// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes Snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
